### ISSUE TITLE
Ability to append tags to builds rather than overwrite

### ISSUE THREAD

**Issue by nwchandler** [colindembovsky/cols-agent-tasks#115]
It appears that the Tag Build functionality currently overwrites tags on a build without an option for appending them. Specifically, Tasks/TagBuild/tagBuild.ts calls [addBuildTags ](https://github.com/Microsoft/azure-devops-node-api/blob/master/api/BuildApi.ts#L3258)from the Azure DevOps node API / build API, which will replace the tags on a build with an input list. If there were a mechanism to optionally call [addBuildTag](https://github.com/Microsoft/azure-devops-node-api/blob/master/api/BuildApi.ts#L3211), it looks like it would simply append a new tag.

Has this been investigated in the past? If there is no reason NOT to include it, would you be open to a PR?

**Comment by colindembovsky** [comment]
Hi @nwchandler - does `addBuildTags` really overwrite?? I would be surprised. If that is the case, then perhaps the solution is to `getBuildTags` first, then append the new tag (if it doesn't exist) and then call `addBuildTag`. Happy to accept PRs!

**Comment by nwchandler** [comment]
Hi, @colindembovsky - Well, it seems that yesterday must have just been a long day or something. I swear I validated that posting to the tag path would overwrite existing tags, and as I was typing up a response to you this morning, I decided to verify one last time. And, it appears I was wrong. I confirmed this morning that posting does NOT overwrite them. I must be overwriting my tags someplace else. 🤦‍♂️ 

Sorry for bugging you, but thanks for the quick response yesterday!
